x-pack/platform/packages/shared/kbn-streams-schema/src/models/ingest/failure_store/index.ts

```diff
@@ -5,11 +5,11 @@
  * 2.0.
  */
 
+import type { IndicesDataStream } from '@elastic/elasticsearch/lib/api/types';
 import { z } from '@kbn/zod';
 import { NonEmptyString } from '@kbn/zod-helpers';
-import type { IndicesDataStream } from '@elastic/elasticsearch/lib/api/types';
 
-export interface FailureStore {
+export interface EffectiveFailureStore {
```

Comment:
I don't think we need that dedicated interface, it should be an union of the existing types minus inherit

```
type EffectiveFailureStore = FailureStoreDisabled | FailureStoreEnabled
```